src/infill.cpp

```diff
@@ -259,6 +259,11 @@ void Infill::addLineInfill(Polygons& result, const PointMatrix& rotation_matrix,
 
 void Infill::generateLineInfill(Polygons& result, int line_distance, const double& fill_angle, int64_t shift)
 {
+    if (infill_origin.X != 0 || infill_origin.Y != 0)
+    {
+        const double fill_angle_rads = fill_angle * M_PI / 180;
+        shift += infill_origin.X * std::cos(fill_angle_rads) - infill_origin.Y * std::sin(fill_angle_rads);
+    }
```

Comment:
Could you make this in to a utility function in linearAlg2D instead? Just rotating a point could be useful elsewhere too.

We have a function called rotateAround() there but that might be too heavyweight for this sort of thing.

Reply:
I don't mind packaging that code into a function but I'm struggling to think of what it should be called! It's not really rotating a point it's doing a calculation whereby the value of shift is adjusted based on the infill origin point and the infill angle.